examples/resources/prefect_block/resource.tf

```diff
@@ -47,3 +47,25 @@ resource "prefect_block" "gcp_credentials_key" {
     "service_account_info" = jsondecode(base64decode(google_service_account_key.test_bot.private_key))
   })
 }
+
+# example:
+# some resources need to be referenced using a "$ref" key
```

Comment:
I'd like to be more specific here so users know when the `$ref` key is needed. Looking into it.

Reply:
References to the application looking for this key:
- https://github.com/PrefectHQ/prefect/blob/2f000f9db4bef6bfc838e3a150168a4c48446045/src/prefect/utilities/templating.py#L246
- https://github.com/PrefectHQ/prefect/blob/2f000f9db4bef6bfc838e3a150168a4c48446045/src/prefect/server/models/block_documents.py#L116
- https://github.com/PrefectHQ/prefect/blob/2f000f9db4bef6bfc838e3a150168a4c48446045/src/prefect/server/api/validation.py#L115-L117
- https://github.com/PrefectHQ/prefect/blob/2f000f9db4bef6bfc838e3a150168a4c48446045/src/prefect/blocks/core.py#L95

Reply:
I was hoping it might be visible in the API docs, such as in [Create block document](https://app.prefect.cloud/api/docs#tag/Block-documents/operation/create_block_document_api_accounts__account_id__workspaces__workspace_id__block_documents__post) - but the `data` field is just described as `object (Data)` and the example payload is just `"data": {}`. So I'm looking to see how and end user would have known to format the data this way.

Reply:
yeah unfortunately i dont think we do a good enough job of documenting the JSON Schema part of the API, especially when it comes to Schema references like this ($ref)

https://json-schema.org/understanding-json-schema/structuring#dollarref

the best way I can illustrate when someone might need to make a reference like this is in the `prefect block type inspect` output:

```
➜ prefect block type inspect dbt-core-operation
```

in here, you'll notice most attributes with some `type`.  but if you look at `dbt_cli_profile`, it has the `$ref` expression:

![image](https://github.com/user-attachments/assets/a7f50b56-061f-4445-9d07-733774227e69)

for the most part, these references are for other Blocks (or rather, block_documents).  so we'd make a `$ref` to the `block_document_id`, which be a `prefect_block` resource's `.id` attribute

Reply:
a picture is worth a thousand words, so if you wanted to include a screenshot in the docs, i want to say we'd have to create a `template` for this page in the docs (kind of like our `templates/guides/getting-started.md`).  I'm a little rusty on the terraform-plugin-docs library when it comes to customizing `resource` doc pages, but i think you can create a custom templates for specific resource files

https://github.com/hashicorp/terraform-plugin-docs?tab=readme-ov-file#conventional-paths

Reply:
definitely not required, though

Reply:
```suggestion
# When referencing other Terraform resources from within the `data` attribute,
# the reference must be nested under a `$ref` key.
# docs: https://json-schema.org/understanding-json-schema/structuring#dollarref
```

Something like this may be more accurate/specific.

Reply:
Oops hadn't refreshed the page, checking your comments now.

Reply:
@parkedwards - nice, thanks for the pointers. Think I got it:

- [Rendered markdown doc](https://github.com/PrefectHQ/terraform-provider-prefect/blob/c28da120c52d06251976707b155dc47df32e07e9/docs/resources/block.md)
- [Image path replaced with PR branch name to confirm path is correct](https://raw.githubusercontent.com/PrefectHQ/terraform-provider-prefect/fix-block-data-update/docs/images/block-type-inspect-ref-key.png)

I created a template file for the Block document page, and added this new section into the template 👍🏼